### PR TITLE
Update documentation for correct curl installation with wolfSSL and HTTP

### PR DESCRIPTION
**Pull request by momuno**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [x] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 
  - [x] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)
This adds documentation for the necessary configuration and installation of curl to resolve issue #1829.  It only pertains to wolfSSL and HTTP used together.

### CODE REVIEW

File: doc/devbox_setup.md [hunk 1]
```diff
@@ -151,6 +151,16 @@ This section describes how to set up a development environment for the C SDK on
   sudo apt-get update
   sudo apt-get install -y git cmake build-essential curl libcurl4-openssl-dev libssl-dev uuid-dev
   ```
+  NOTE: If you are planning to use HTTP with wolfSSL, you must instead configure curl before installation.
```
Comment (ericwol-msft) [inline doc/devbox_setup.md:154]:
```suggestion
  NOTE: If you are planning to use HTTP with wolfSSL, you must instead curl before installation.
```

File: doc/devbox_setup.md [hunk 2]
```diff
@@ -151,6 +151,16 @@ This section describes how to set up a development environment for the C SDK on
...
+  1. Download the latest [curl](https://github.com/curl/curl/releases).
+  2. After extracting, enter the directory and run the following commands:
```
Comment (ericwol-msft) [inline doc/devbox_setup.md:156]:
```suggestion
  2. After extracting, change the working directory and run the following commands:
```

**Review by ericwol-msft — APPROVED** [review]

**Review by ericwol-msft — APPROVED** [review]

File: doc/devbox_setup.md [hunk 3]
```diff
@@ -151,6 +151,9 @@ This section describes how to set up a development environment for the C SDK on
   sudo apt-get update
   sudo apt-get install -y git cmake build-essential curl libcurl4-openssl-dev libssl-dev uuid-dev
   ```
+  NOTE: If you are planning to use HTTP with wolfSSL, you must configure curl before installation.
+  1. Download the latest [curl](https://github.com/curl/curl/releases).
+  2. See curl [documentation](https://curl.se/docs/install.html) to use the wolfSSL option for configuration.  Install.
```
Comment (hihigupt) [inline doc/devbox_setup.md:156]: few other things that can be added in this is 
1. Curl support with wolfssl
https://www.wolfssl.com/products/curl/
2. Which options are required by azure from wolfssl  lib : renegotiation

Comment (momuno) [reply doc/devbox_setup.md:156]: good idea.  I'll do another PR since this is merged.

**Pull request merged**